txp/Pos/Binary/Txp/Core.hs

```diff
@@ -31,17 +33,19 @@ instance Bi T.TxIn where
     encode (T.TxInUtxo utxoTxIn) =
         encodeListLen 2 <>
         encode (0 :: Word8) <>
-        encode (serialize' utxoTxIn)
+        encodeCborDataItem (serialize' utxoTxIn)
     encode (T.TxInUnknown tag bs) =
         encodeListLen 2 <>
         encode tag <>
-        encode bs
+        encodeCborDataItem bs
     decode = do
         enforceSize "TxIn" 2
         tag <- decode @Word8
         case tag of
-            0 -> T.TxInUtxo <$> decode
-            _ -> T.TxInUnknown tag <$> decode
+            0 -> T.TxInUtxo <$> decodeCborDataItem
+            _ -> do
+                decodeCborDataItemTag
+                T.TxInUnknown tag <$> decode
```

Comment:
This can be the `decodeUnknownCborDataItem` and then you don't need to use or export `decodeCborDataItemTag` since you'll make sure that is handled within `decodeUnknownCborDataItem`.

    _ -> T.TxInUnknown tag <$> decodeUnknownCborDataItem